src/lib.rs

```diff
@@ -53,25 +53,16 @@ let x: u32 = match Ok::<_, Never>(42) {
 #![no_std]
 #![forbid(unsafe_code)]
 
-/// Workaround for `fn_traits` and/or `unboxed_closures`
 mod fn_traits {
-    pub
-    trait FnOnce<Args> {
+    pub trait FnOutput {
```

Comment:
Naming nit: I'd rather we not have `Output` mentioned twice w.r.t. the assoc type, so what about using `FnPtr` here?

```suggestion
    pub trait FnPtr {
```